File: ros_ws/src/intro_pkg1/scripts/ServiceServer1.py
#!/usr/bin/env python
# Import files
import rospy
from intro_pkg1.srv import *

# Service server function
def equation(request):
    # Response object
    response = FloatIOResponse()
    # Relation between request and response
    response.output = 2 * (request.input) + 1
    rospy.loginfo("Request: {0}, response: {1}".format(request.input, response.output))
    # Return the response object
    return response

def main():
    # Initialize the node
    rospy.init_node("Equation_server", anonymous=True)
    # Setup the server
    rospy.Service("equation_service", FloatIO, equation)
    rospy.loginfo("FloatIO service activated. Equation: Y = 2 * X + 1, Y is output and X is input")
    # Spin the node
    rospy.spin()

# Main code that will be executed
if __name__ == "__main__":
    try:
        # Main function
        main()
    except rospy.ROSInterruptException:
        rospy.logfatal("Node crashed")

File: ros_ws/src/src/intro_pkg1/src/ServiceServer1.cpp
// Include header files
#include "ros/ros.h"
// The service header file
#include "intro_pkg1/FloatIO.h"

// Definition function
bool equation(intro_pkg1::FloatIO::Request &req, intro_pkg1::FloatIO::Response &res) {
    // Relation between output and input
    res.output = 2 * (req.input) + 1;
    ROS_INFO("Request: [%f], Response: [%f]", req.input, res.output);
    return true;
}

int main(int argc, char **argv)
{
    // Initialize the Node, name is "Server_FloatIO"
    ros::init(argc, argv, "Server_FloatIO");
    // Node Handler
    ros::NodeHandle nodeHandler;
    // Service server
    ros::ServiceServer serviceHandler = nodeHandler.advertiseService("equation_service", equation);
    ROS_INFO("Service setup complete. Equation is Y = 2 * X + 1, Y is output and X is input.");
    // Go into listening mode, spin the node
    ros::spin();
    return 0;
}
